lib/Canonicalize/MakeModelGEPPass.cpp

```diff
@@ -472,8 +471,9 @@ struct IRArithmetic {
   // Factories
   //
 
-  static IRArithmetic address(model::QualifiedType Type, Value *Address) {
-    return IRArithmetic(ModelTypedIRAddress(dropPointer(Type), Address));
+  static IRArithmetic address(const model::Type &T, Value *Address) {
+    return IRArithmetic(ModelTypedIRAddress(model::copyType(T.stripPointer()),
+                                            Address));
```

Comment:
Maybe I asked about this on the other PR too, but what does `model::copyType` do?

Reply:
It's a TTG-generated helper for painlessly converting `X` to `UpcastableX`.

```
inline UpcastableType copyType(const Type &From) {
  UpcastableType Result;
  upcast(&From, [&Result]<typename T>(const T &Upcasted) {
    Result = UpcastableType::make<T>(Upcasted);
  });
  return Result;
}
```

Reply:
Can't this be the body of the copy constructor of `UpcastableType`?

Reply:
I have a lot of FUD about such non-obvious "copy" constructor, but technically it should work.

Reply:
I feel the same, but I want to push hard for model data structures to appear as natural as they can, we should minimize the aspects that make them "our private" data structures, that have their weirdness and corner cases and "touch with care".

That's why I insist so much about giving it a `std::variant` look and feel. Would you want `std::copyVariant` instead of using the copy constructor of `variant`? Likely not.

Reply:
Done.